src/sentry/scripts/utils/locking/delete_lock.lua

```diff
@@ -0,0 +1,12 @@
+local key = KEYS[1]
+local uuid = ARGV[1]
+
+local value = redis.call('GET', key)
+if not value then
+    return redis.error_reply(string.format("No lock at key exists at key: %s", key))
+elseif value ~= uuid then
+    return redis.error_reply(string.format("Lock at %s was set by %s, and cannot be released by %s.", key, value, uuid))
+else
+    redis.call('DEL', key)
```

Comment:
> it'll just return `0` if no keys were deleted.

This situation shouldn't happen because the `GET` ensures the key exists, and the script will otherwise error if it doesn't exist.

> could/should we just return `redis.status_reply('OK')`?

No strong opinion here, I'm happy to change it. It gets returned to the client as a string:

```python
>>> r.eval('return redis.status_reply("OK")', 0, ())
'OK'
```

We don't really use this return value for anything meaningful, and I think that given the script is called `delete_lock` an `OK` response probably makes enough sense.